File: Atcoder_#337/ruby/b_extended_abc.rb
s = gets.chomp
s_chars = s.chars

result = "Yes"

s.length.times do |i|
  if s_chars[i-1] == "B" && s_chars[i] == "A"
    result = "No"
    break
  elsif s_chars[i-1] == "C" && s_chars[i] == "A" || s_chars[i] == "B"
    result = "No"
    break
  elsif s_chars[i-1] == "B" || s_chars[i-1] == "C"
    result = "No"
    break
  else
    next
  end
end

puts result

File: Atcoder_#337/javascript/b_extended_abc.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on('line', (input) => {
    const s_chars = input.split('');
    let result = "Yes";

    for (let i = 1; i < s_chars.length; i++) {
        if (s_chars[i - 1] === "B" && s_chars[i] === "A") {
            result = "No";
            break;
        } else if (s_chars[i - 1] === "C" && s_chars[i] !== "C") {
            result = "No";
            break;
        }
    }

    console.log(result);
    rl.close();
});
